docs/scaling.md

```diff
@@ -0,0 +1,190 @@
+# Scaling DataPlane
+
+## Scaling Out
+
+Scale out of an existing dataplane with more edpm nodes can be achieved by adding new
+nodes to the `nodes` section of the `OpenStackDataPlaneNodeSet` spec. Ensure that
+there are enough BMHs in `Available` state in the required namespace with the desired
+labels for baremetal nodes.
+
+Pre-Provisioned:
+
+    apiVersion: dataplane.openstack.org/v1beta1
+    kind: OpenStackDataPlaneNodeSet
+    metadata:
+      name: openstack-edpm
+    spec:
+      preProvisioned: True
+      nodes:
+      ...
+        edpm-compute-2:
+          hostName: edpm-compute-2
+          ansible:
+            ansibleHost: 192.168.122.102
+          networks:
+          - name: CtlPlane
+            subnetName: subnet1
+            defaultRoute: true
+            fixedIP: 192.168.122.102
+          - name: InternalApi
+            subnetName: subnet1
+          - name: Storage
+            subnetName: subnet1
+          - name: Tenant
+            subnetName: subnet1
+      ...
+
+Baremetal:
+
+    apiVersion: dataplane.openstack.org/v1beta1
+    kind: OpenStackDataPlaneNodeSet
+    metadata:
+      name: openstack-edpm
+    spec:
+      nodes:
+      ...
+        edpm-compute-2:
+          hostName: edpm-compute-2
+      ...
+
+To deploy on the additional nodes, a new `OpenStackDataPlaneDeployment` CR should be
+created with the `OpenStackDataPlaneNodeSet` in the `nodeSets` section.
+
+Once deployment completes `OpenStackDataPlaneNodeSet` would be ready as shown below.
```

Comment:
Wouldn't deployment only be complete after `OpenStackDataPlaneDeployment` is ready?

Reply:
Once `OpenStackDataPlaneDeployment` is Ready (i.e "SetupComplete" ),  we set the `OpenStackDataPlaneNodeSet` `DeploymentReadyCondition=True` [1] that would set it to `Ready` (i.e "NodeSet Ready").

[1] https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/controllers/openstackdataplanenodeset_controller.go#L303-L316

Reply:
Ack